### ISSUE TITLE
Use Webform remote posts to try and support Prospector app needs

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#8103]
# Story

As an owner of the Prospector application, we would like SiteNow webforms to be able to pass information to the Prospector application via middleware so that we can better recruit potential prospects.

# Additional Info

AIS has a Prospector App where they keep information about potential prospects. They would like webforms on some SiteNow sites to be able to post to a middleware endpoint so that this information can be captured in the Prospector App. The College of Education is their first priority and there would be multiple different forms (40+) on this site that would use this functionality. Eventually, this would extend to other Collegiate sites, Admissions, etc.

Sounds like each form would need some sort of hidden field that would contain a UUID that would need to pass to the endpoint along with the included form fields.

Basic authentication needed for the endpoint.

Test: https://test.its.uiowa.edu/prospect-api/api/prospect/submit

Scope of this would be to work with AIS to get a single Education webform posting to their middleware using basic auth, passing in a UUID and submitted form values (firstName, lastName, email, siteInteractionUuid).

We can have future slices to improve this, make it more widely available and to document or make it easier for webmasters to set this up without our team having to be involved with each form.

**Comment by briand44** [comment]
Grooming... We have a feature split (advanced webform), that we can currently activate through the UI. Enabling the feature split doesn't import the config. A developer would need to import the config. The advanced webform feature split currently only contains this functionality. Currently Tippie and UIP have this enabled. AIS should be getting us an endpoint, UUID, etc. for us to begin testing integrating during this next sprint. Let's familiarize ourselves, bring up any concerns and see if we can support their needs. Seems like this feature could work with an authenticated endpoint based on https://www.drupal.org/project/webform/issues/3056017.

**Comment by briand44** [comment]
> We deployed middleware app in both test and prod. Please see following details for test end point
>  
> Test: https://test.its.uiowa.edu/prospect-api/api/prospect/submit
>  
> It’s a POST request which accepts json and at a minimum this endpoint accepts firstName, lastName, email and siteInteractionUuid (a hidden property) and the value will be different for each interaction(I will send you the value separately for testing).
>  
> At this point, this endpoint is protected with basic authentication. I will send you credentials separately.

**Comment by pyrello** [comment]
I was able to get the webform remote posting (via the `advanced_webform` feature) to the Prospector app endpoint (middleware) working. This was done entirely through configuration of an existing form on a local version of the education website. At this stage there is no PR because there is nothing to capture in terms of changes that could be added to the `advanced_webform` feature. We have discussed next steps in #8138 and work will continue there.
